### ISSUE TITLE
iceStateChange not called when onconnectionstatechange is invoked first

### ISSUE THREAD

**Issue by samuelmaddock** [feross/simple-peer#692]
Browser: Chrome 83.0.4103.116

## Problem

In my application, I'd like to react to when a `failed` state change occurs to try and reconnect a user.

Looking at the chrome://webrtc-internals logs reveals the following event order:
![chrome_2020-06-26_01-28-08AhIp](https://user-images.githubusercontent.com/1656324/85824070-ec6a2c00-b74c-11ea-9654-a2fc23fee6f3.png)

`iceconnectionstatechange` is logged, then `connectionstatechange`. However, these callbacks seem to be invoked in the reverse order for the simple-peer `RTCPeerConnection` instance.

![chrome_2020-06-26_01-37-37MXcD](https://user-images.githubusercontent.com/1656324/85824385-ae213c80-b74d-11ea-9e84-e49c8cc24bda.png)

The `SimplePeer` instance is subsequently destroyed and no callback to `iceconnectionstatechange` is invoked. My intentions were to capture the `failed` state in an `iceStateChange` listener, but it's never called.

## Proposal

Since the webrtc-internal tool says `iceconnectionstatechange` is a legacy event, I'd prefer to be able to listen for `stateChange` instead of `iceStateChange`.

A new event `stateChange` should be emitted when the `connectionstatechange` callback is invoked.

Additionally, both `stateChange` and `iceStateChange` would ideally be called prior to the `SimplePeer` instance being destroyed. Maybe the call to `SimplePeer.destroy()` could be delayed by one tick?
```diff
_onConnectionStateChange () {
  if (this.destroyed) return
  if (this._pc.connectionState === 'failed') {
-    this.destroy(makeError('Connection failed.', 'ERR_CONNECTION_FAILURE'))
+    setTimeout(() => {
+      this.destroy(makeError('Connection failed.', 'ERR_CONNECTION_FAILURE'))
+    }, 0)
  }
}
```
This feels a bit dirty though so I'm unsure.

**Comment by samuelmaddock** [comment]
Looking more into this, the `onconnectionstatechange` event handler is not implemented as of Firefox 77.0.1. Because of this, I think we should just ensure that the `iceStateChange` event is always fired prior to destroying the peer instance.

**Comment by samuelmaddock** [comment]
Looking closer, I think this might be a Chrome bug. The non-legacy `iceconnectionstatechange` event is never called when the state changes to `failed`.

![chrome_2020-06-26_02-15-24qha8](https://user-images.githubusercontent.com/1656324/85826747-ef681b00-b752-11ea-8bb8-fe57f8e0b4b2.png)

Removing the `destroy()` call in `_onConnectionStateChange()` doesn't make `_onIceStateChange` get called either.

**Comment by t-mullen** [comment]
Your proposal is reasonable. We can solve this type of bug in a more general way if we put the peer in a "destroying" state when `_destroy()` is called, then transition to "destroyed" on the next microtask.

**Comment by Mupati** [comment]
I have the same error as @samuelmaddock shared above. Everything works fine one the same network but on different networks after the `type: answer` is received by the initiator, the connection terminates. I get this error ```Error: Connection failed.
    at h (vendor.3a80edfa.js:15)
    at p._onConnectionStateChange (vendor.3a80edfa.js:15)
    at RTCPeerConnection._pc.onconnectionstatechange (vendor.3a80edfa.js:15)```

**Comment by feross** [comment]
Released as 9.9.2
